bin/ignore-extraneous-style-assets.js

```diff
@@ -0,0 +1,59 @@
+/**
+ * Webpack plugin to prevent extraneous assets from being emitted for stylesheets.
+ *
+ * JS files and PHP asset files generated by `DependencyExtractionWebpackPlugin` are currently ignored.
+ */
+class IgnoreExtraneousStyleAssets {
+	isCssFile( file ) {
+		return /^(?!!!).+\.(?:sc|sa|c)ss$/.test( file );
+	}
+
+	generateIgnoreRegex( entries ) {
+		const expression = entries
+			.map( ( entry ) => [ `${ entry }.js`, `${ entry }.asset.php` ] )
+			.flat()
+			.join( '|' );
```

Comment:
I guess this could be done in a single pass if `reduce()` was used:

```suggestion
		const expression = entries
			.reduce( ( acc, entry ) => `${ acc }|${ entry }.js|${ entry }.asset.php`, '' )
			.substr( 1 );
```

Anyway, it works well as is so no need to change.